File: FTTradeDemo/FTTradeDemo3/FTTradeDemo3/FTTradeCardView.m
//
//  FTTradeCardView.m
//  FTTradeDemo3
//
//  Created by admin on 2022/4/8.
//

#import "FTTradeCardView.h"

@interface FTTradeCardView()

@property(nonatomic,strong)UILabel *topTitle;
@property(nonatomic,strong)UILabel *bottomTitle;
@property(nonatomic,strong)UIImageView *cardImage;

@end

@implementation FTTradeCardView
- (instancetype)initWithCard:(Card *)cardModel {
    self = [super init];
    
    if(self){
        self.cardModel = cardModel;
        
        [self setView];
        [self addSubview:self.cardImage];
        [self addSubview:self.topTitle];
        [self addSubview:self.bottomTitle];
    }
    return self;
}

- (void)layoutSubviews {
    [super layoutSubviews];
 
}

- (void)onSelected {
    if(self.SelectedBlock){
        self.SelectedBlock();
    }
}

- (void)setView {
    UIImage *tmpImage = [UIImage imageNamed:self.cardModel.cardContentUrl];
    self.cardImage.image = tmpImage;
    [self.cardImage setUserInteractionEnabled:YES];
    [self.cardImage setMultipleTouchEnabled:YES];
    [self.cardImage addGestureRecognizer:[[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(onSelected)]];
    
    self.topTitle.text = self.cardModel.cardTitle;
    self.topTitle.adjustsFontSizeToFitWidth = YES;
    self.bottomTitle.text = self.cardModel.cardBottomTitle;
    self.bottomTitle.adjustsFontSizeToFitWidth = YES;
}

- (UIImageView *)cardImage {
    if(!_cardImage){
        _cardImage = [[UIImageView alloc] initWithImage:[UIImage imageNamed:self.cardModel.cardContentUrl]];
    }
    return _cardImage;
}
- (UILabel *)topTitle {
    if(!_topTitle){
        _topTitle = [[UILabel alloc] initWithFrame:CGRectMake(1, 1, 15, 15)];
    }
    return _topTitle;
}
- (UILabel *)bottomTitle {
    if(!_bottomTitle){
        _bottomTitle =[[UILabel alloc] initWithFrame:CGRectMake(100, 100, 100, 100)];
    }
    return _bottomTitle;
}
@end

File: FTTradeDemo/FTTradeSwiftDemo6/FTTradeSwiftDemo6/FTTradeCardView.swift
//
//  FTTradeCardView.swift
//  FTTradeSwiftDemo6
//
//  Created by admin on 2022/4/20.
//

import Foundation
import UIKit

class FTTradeCardView : UIView {
    
    typealias cardViewBlock = () -> Void
    var cardViewCallBackBlock : cardViewBlock?
    
    lazy private var topTitleLabel: UILabel = {
        let tmplabel = UILabel(frame: CGRect(x: 1, y: 1, width: 15, height: 15))
        tmplabel.sizeToFit()
        return tmplabel
    }()
    
    lazy private var bottomTitleLabel : UILabel = {
        let tmplabel = UILabel(frame: CGRect(x: 100, y: 140, width: 15, height: 15))
        tmplabel.sizeToFit()
        return tmplabel
    }()
    lazy private var cardImage : UIImageView = {
        let tmpImage = UIImageView(image: UIImage(named: "card_before.png"))
        tmpImage.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(cardImageClickAction)))
        tmpImage.isUserInteractionEnabled = true
        tmpImage.sizeToFit()
        return tmpImage
    }()
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        self.addSubview(cardImage)
        self.addSubview(topTitleLabel)
        self.addSubview(bottomTitleLabel)
    }
    
    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    func setCardView(cardModel: FTTradeCard) {
        let tmpImage = UIImage(named: cardModel.cardContentUrl)
        cardImage.image = tmpImage
        
        topTitleLabel.text = cardModel.cardTitle
        bottomTitleLabel.text = cardModel.cardTitle
    }
    
    @objc func cardImageClickAction() {
        print("click")
        if let back = cardViewCallBackBlock {
            back()
        }
    }

}

